lib/MetaCPAN/Role/Script.pm

```diff
@@ -195,18 +254,131 @@ before run => sub {
+                    = { 'index_name' => $1, 'health' => $2 };
+
+                $ihealth = 0 if ( $2 eq 'red' );
+            }
+        }
+        else {
+            foreach ( keys %{ $self->indices_info } ) {
+                $ihealth = 0
+                    if ( $self->indices_info->{$_}->{'health'} eq 'red' );
+            }
+        }
+    }
+
+    if ($ihealth) {
+        if ( $irefresh || scalar( keys %{ $self->aliases_info } ) == 0 ) {
+            my $sinfo_rs
+                = $self->es->cat->aliases( h => [ 'alias', 'index' ] );
+
+            $self->aliases_info({});
+
+            while ( $sinfo_rs =~ /^([^[:space:]]+) +([^[:space:]]+)/gm ) {
+                $self->aliases_info->{$1}
+                    = { 'alias_name' => $1, 'index' => $2 };
+            }
+        }
+
+        $ihealth = 0 if ( scalar( keys %{ $self->aliases_info } ) == 0 );
+    }
+
+    return $ihealth;
+}
+
+sub await {
+    my $self   = $_[0];
+    my $iready = 0;
+
+    if ( scalar( keys %{ $self->cluster_info } ) == 0 ) {
+        my $es       = $self->es;
+        my $iseconds = 0;
+
+        log_info {"Awaiting Elasticsearch ..."};
+
+        do {
+            eval {
+                $iready = $es->ping;
+
+                if ($iready) {
+                    log_info {
+                        "Awaiting $iseconds / "
+                            . $self->arg_await_timeout
+                            . " : ready"
+                    };
+
+                    $self->cluster_info( \%{ $es->info } );
+                }
+            };
+
+            if ($@) {
+                if ( $iseconds < $self->arg_await_timeout ) {
+                    log_info {
+                        "Awaiting $iseconds / "
+                            . $self->arg_await_timeout
+                            . " : unavailable - sleeping ..."
+                    };
+
+                    sleep(1);
+
+                    $iseconds++;
+                }
+                else {
+                    log_error {
+                        "Awaiting $iseconds / "
+                            . $self->arg_await_timeout
+                            . " : unavailable - timeout!"
+                    };
+
+                    #Set System Error: 112 - EHOSTDOWN - Host is down
+                    $self->exit_code(112);
+                    $self->handle_error($@, 1);
+                }
+            }
+        } while ( !$iready && $iseconds <= $self->arg_await_timeout );
+    }
+    else {
+        #ElasticSearch Service is available
+        $iready = 1;
+    }
+
+    return $iready;
+}
+
 sub are_you_sure {
     my ( $self, $msg ) = @_;
 
-    if (is_interactive) {
-        print colored( ['bold red'], "*** Warning ***: $msg" ), "\n";
-        my $answer = prompt
-            'Are you sure you want to do this (type "YES" to confirm) ? ';
-        if ( $answer ne 'YES' ) {
-            print "bye.\n";
-            exit 0;
+    unless ( $self->arg_yes ) {
```

Comment:
When the operator intents to interrupt the process it will only skip the index elimination but the whole indexing process will continue since the application returns `[ 0 ]` which usually indicates success:
```
$ docker-compose exec api index-cpan.sh
2021-12-18 17:58:28 I index-cpan.sh: Indices re-creating ...
2021/12/18 17:58:30 I mapping: Awaiting Elasticsearch ...
2021/12/18 17:58:30 I mapping: Awaiting 0 / 15 : ready
*** Warning ***: this will delete EVERYTHING and re-create the (empty) indexes
Are you sure you want to do this (type "YES" to confirm) ? 2021-12-18 17:58:42 I index-cpan.sh: Re-creation finished with [0]
2021-12-18 17:58:42 I index-cpan.sh: Re-creation Log:
''
2021-12-18 17:58:42 I index-cpan.sh: ElasticSearch - Info collecting ...
2021-12-18 17:58:43 I index-cpan.sh: Info finished with [0]
2021-12-18 17:58:43 I index-cpan.sh: Info Log:
'{
   "indices_info" : {
      "contributor" : {
         "health" : "yellow",
         "index_name" : "contributor"
      },
      "user" : {
         "index_name" : "user",
         "health" : "yellow"
      },
      "cpan_v1_01" : {
         "index_name" : "cpan_v1_01",
         "health" : "yellow"
      },
      "cover" : {
         "health" : "yellow",
         "index_name" : "cover"
      }
   },
   "aliases_info" : {
      "cpan" : {
         "alias_name" : "cpan",
         "index" : "cpan_v1_01"
      }
   },
   "cluster_info" : {
      "tagline" : "You Know, for Search",
      "cluster_uuid" : "-lNau9OdTAqAUBa_SLbtbw",
      "version" : {
         "lucene_version" : "5.5.4",
         "build_timestamp" : "2017-07-18T12:17:44Z",
         "build_snapshot" : false,
         "number" : "2.4.6",
         "build_hash" : "5376dca9f70f3abef96a77f4bb22720ace8240fd"
      },
      "cluster_name" : "elasticsearch",
      "name" : "Cerebra"
   }
}'
2021-12-18 17:58:43 I index-cpan.sh: Packages downloading ...
receiving incremental file list
authors/

sent 34 bytes  received 2,998 bytes  866.29 bytes/sec
total size is 1,653,996  speedup is 545.51
receiving incremental file list
# [...]
```
If this function is meant to interrupt the process it should return an **Error Exit Code**.
And ideally it also would give a more conclusive message.